specutils/manipulation/resample.py

```diff
@@ -0,0 +1,293 @@
+        ----------
+        x : ndarray
+            The input dispersion axis values.
+
+        Returns
+        -------
+        edges : ndarray
+            Calcualated bin edges, including left and right most bin edges.
+        """
+        inside_edges = (x[1:] + x[:-1]) / 2
+        edges = np.insert(inside_edges, 0, 2 * x[0] - inside_edges[0])
+        edges = np.append(edges, 2 * x[-1] - inside_edges[-1])
+
+        return edges
+
+
+class FluxConservingResample(ResampleBase):
+    """
+    This resample algorithim conserves overall flux (as opposed to flux density).
+    Algorithim based on the equations documented in the following paper:
+    https://ui.adsabs.harvard.edu/abs/2017arXiv170505165C/abstract
+    """
+
+    def __call__(self, orig_spectrum, fin_lamb):
+        """
+        Return the resulting `~specutils.Spectrum1D` of the resampling.
+        """
+        return self.resample1D(orig_spectrum, fin_lamb)
+
+    def _resample_matrix(self, orig_lamb, fin_lamb):
+        """
+        Create a re-sampling matrix to be used in re-sampling spectra in a way
+        that conserves flux. This is adapted from
+        https://ui.adsabs.harvard.edu/abs/2017arXiv170505165C/references,
+        eprint arXiv:1705.05165. This code was heavily influenced by Nick Earl's
+        resample rough draft.
+
+        Parameters
+        ----------
+        orig_lamb : ndarray
+            The original dispersion array.
+        fin_lamb : ndarray
+            The desired dispersion array.
+
+        Returns
+        -------
+        resample_mat : ndarray
+            An [[N_{fin_lamb}, M_{orig_lamb}]] matrix.
+        """
+        # Lower bin and upper bin edges
+        orig_edges = self._calc_bin_edges(orig_lamb)
+        fin_edges = self._calc_bin_edges(fin_lamb)
+
+        # I could get rid of these alias variables,
+        # but it does add readability
+        orig_low = orig_edges[:-1]
+        fin_low = fin_edges[:-1]
+        orig_upp = orig_edges[1:]
+        fin_upp = fin_edges[1:]
+
+        # Here's the real work in figuring out the bin overlaps
+        # i.e., contribution of each original bin to the resampled bin
+        l_inf = np.where(orig_low > fin_low[:, np.newaxis],
+                         orig_low, fin_low[:, np.newaxis])
+        l_sup = np.where(orig_upp < fin_upp[:, np.newaxis],
+                         orig_upp, fin_upp[:, np.newaxis])
+
+        resamp_mat = (l_sup - l_inf).clip(0)
+        resamp_mat *= (orig_upp - orig_low)
+
+        # set bins that don't overlap 100% with original bins
+        # to zero by checking edges, and applying generated mask
+        left_clip = np.where(fin_edges[:-1] - orig_edges[0] < 0, 0, 1)
+        right_clip = np.where(orig_edges[-1] - fin_edges[1:] < 0, 0, 1)
+        keep_overlapping_matrix = left_clip * right_clip
+
+        resamp_mat *= keep_overlapping_matrix[:, np.newaxis]
+
+        return resamp_mat
+
+    def resample1D(self, orig_spectrum, fin_lamb):
```

Comment:
```suggestion
    def resample1d(self, orig_spectrum, fin_lamb):
```
PEP8 is admittedly a bit ambiguous, but in general the preference I tend to see is "1d" over "1D" in places that are normally all-lowercase.

(Note that I didn't go through and find all of these - probably it's easier done as a global find-and-replace in your editor)